src/getExtensionApi.ts

```diff
@@ -15,3 +17,22 @@ export async function getResourceGroupsApi(): Promise<AzureHostExtensionApi> {
         throw new Error(localize('noResourceGroupExt', 'Could not find the Azure Resource Groups extension'));
     }
 }
+
+/**
+ * @param installMessage Override default message shown if extension is not installed.
+ */
+export async function getAzureContainerAppsApi(context: IActionContext, installMessage?: string): Promise<acaApi.AzureContainerAppsExtensionApi> {
```

Comment:
You should call this way earlier in the create containerized function app flow. You get through creating every resource and then it fails at the end.